src/atdf/chip.rs

```diff
@@ -20,11 +20,20 @@ pub fn parse(el: &xmltree::Element) -> crate::Result<chip::Chip> {
     .map(|p| (p.name.clone(), p))
     .collect();
 
+    let interrupts = device
+        .first_child("interrupts")?
+        .children
+        .iter()
+        .map(atdf::interrupt::parse)
+        .map(|r| r.map(|int| (int.name.clone(), int)))
+        .collect::<Result<_, _>>()?;
```

Comment:
Try if you can remove the turbofish from `.collect()`.  If not, thats ok, but if possible, is would de-clutter the code a little.